### PR TITLE
#2680: EMP initial integration

### PR DESCRIPTION
**Pull request by lat9**
Initial full-on incorporation into core.

### CODE REVIEW

File: admin/customers.php [hunk 1]
```diff
@@ -1370,8 +1370,9 @@ function init() {
                   $contents[] = array('align' => 'text-center', 'text' => '<br><button type="submit" class="btn btn-warning">' . IMAGE_RESET_PWD . '</button> <a href="' . zen_href_link(FILENAME_CUSTOMERS, zen_get_all_get_params(array('cID', 'action')) . 'cID=' . $cInfo->customers_id) . '" class="btn btn-default" role="button">' . IMAGE_CANCEL . '</a>');
                   break;
                 default:
-                  if (isset($_GET['search']))
+                  if (isset($_GET['search'])) {
                     $_GET['search'] = zen_output_string_protected($_GET['search']);
+                  }
```
Comment (scottcwilson) [inline admin/customers.php]: one liner maybe?

File: admin/customers.php [hunk 2]
```diff
@@ -1396,17 +1414,17 @@ function init() {
...
                       $last_order = array(
                         'date_purchased' => $customers_orders->fields['date_purchased'],
                         'order_total' => $customers_orders->fields['order_total'], 
                         'currency' => $customers_orders->fields['currency'], 
                         'currency_value' => $customers_orders->fields['currency_value'],
                       );
-                      foreach ($customers_orders as $result) {
-                        $lifetime_value += ($result['order_total'] * $result['currency_value']);
+                      foreach($customers_orders as $result) {
+                         $lifetime_value+= ($result['order_total'] * $result['currency_value']);
```
Comment (scottcwilson) [inline admin/customers.php]: seems like your editor added this.

File: admin/customers.php (hunk 1 above)
Comment (lat9) [reply admin/customers.php]: FWIW, except for the case of checking for undefined constants, my preference for processing conditionals is to use the braces to help identify the full clause.  I can't count the number of times where I've missed one of those one-liners when trying to reformat a module.

File: admin/customers.php (hunk 2 above)
Comment (lat9) [reply admin/customers.php]: It might have, but what's the issue with those additional spaces?

Comment (lat9) [reply admin/customers.php]: Argh, that must have been Beyond Compare "helping".

**Pull request merged**